cpp/include/raft/linalg/eig.cuh

```diff
@@ -112,6 +114,117 @@ void eigJacobi(const raft::handle_t& handle,
 {
   detail::eigJacobi(handle, in, n_rows, n_cols, eig_vectors, eig_vals, stream, tol, sweeps);
 }
+
+/**
+ * @brief eig decomp with divide and conquer method for the column-major
+ * symmetric matrices
+ * @tparam ElementType the data-type of input and output
+ * @tparam IntegerType Integer used for addressing
+ * @param handle raft::handle_t
+ * @param in input raft::device_matrix_view (symmetric matrix that has real eig values and
+ * vectors)
+ * @param eig_vectors: eigenvectors output of type raft::device_matrix_view
+ * @param eig_vals: eigen values output of type raft::device_vector_view
+ */
+template <typename ElementType, typename IndexType = std::uint32_t>
+void eig_dc(const raft::handle_t& handle,
+            const raft::device_matrix_view<ElementType, IndexType, raft::col_major> in,
+            raft::device_matrix_view<ElementType, IndexType, raft::col_major> eig_vectors,
+            raft::device_vector_view<ElementType, IndexType> eig_vals)
+{
+  RAFT_EXPECTS(in.is_exhaustive(), "Input must be contiguous");
+  RAFT_EXPECTS(eig_vectors.is_exhaustive(), "Eigen Vectors must be contiguous");
+  RAFT_EXPECTS(eig_vals.is_exhaustive(), "Eigen Values must be contiguous");
+  RAFT_EXPECTS(in.size() == eig_vectors.size(), "Size mismatch between Input and Eigen Vectors");
+  RAFT_EXPECTS(eig_vals.size() == in.extent(1), "Size mismatch between Input and Eigen Values");
+
+  eigDC(handle,
+        in.data_handle(),
+        in.extent(0),
+        in.extent(1),
+        eig_vectors.data_handle(),
+        eig_vals.data_handle(),
+        handle.get_stream());
+}
+
+/**
+ * @brief eig decomp to select top-n eigen values with divide and conquer method
+ *        for the column-major symmetric matrices
+ * @tparam ElementType the data-type of input and output
+ * @tparam IntegerType Integer used for addressing
+ * @param handle raft::handle_t
+ * @param in input raft::device_matrix_view (symmetric matrix that has real eig values and
+ * vectors)
+ * @param n_eig_vals: number of eigenvectors to be generated
+ * @param eig_vectors: eigenvectors output of type raft::device_matrix_view
+ * @param eig_vals: eigen values output of type raft::device_vector_view
+ */
+template <typename ElementType, typename IndexType = std::uint32_t>
+void eig_dc_select(const raft::handle_t& handle,
+                   const raft::device_matrix_view<ElementType, IndexType, raft::col_major> in,
+                   std::size_t n_eig_vals,
+                   raft::device_matrix_view<ElementType, IndexType, raft::col_major> eig_vectors,
+                   raft::device_vector_view<ElementType, IndexType> eig_vals,
+                   EigVecMemUsage memUsage)
+{
+  RAFT_EXPECTS(in.is_exhaustive(), "Input must be contiguous");
```

Comment:
Same wrt exhaustive vs row-/col-major. This applies to the rest of the PR so I won't bother pointing out each and every instance.